### PR TITLE
[#8][das#208] Enhance the Terraform script to incorporate the necessary requirements for provisioning servers

### LINKED ISSUE #8: Enhance the Terraform script to incorporate the necessary requirements for provisioning servers

### PR DESCRIPTION
**Pull request by levisingularity**
Closes #8

### CODE REVIEW

File: install-mongodb.sh [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
+   dockerd > /dev/null 2>&1 < /dev/null &
+   docker run hello-world
+   docker rm $(docker ps -a | grep hello | cut -d" " -f1)
+   docker rmi hello-world
+}
+
+function mongo_setup() {
+   docker image pull mongo:$MONGO_VERSION
+
+   docker run --name mongo --restart always -p ${MONGO_PORT}:27017 -d mongo:$MONGO_VERSION
```
Comment (andre-senna) [inline install-mongodb.sh:38]: Please remove defaults for environment variables. If we make our setup based in environment variables we need to make sure they are being set properly. Any unset variable show make the script break with a proper error message.

File: install-mongodb.sh [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
+}
+
+user_setup
+docker_setup
+mongo_setup
```
Comment (andre-senna) [inline install-mongodb.sh:43]: I didn't see where MongoDB port is being set open in ufw. Are we doing it somewhere else?
Anyway, this port should not be open for every IP, but only for the OpenFaas server.

File: install-redis.sh [hunk 3]
```diff
@@ -1,8 +1,132 @@
...
+    create_redis_conf
+
+    docker image pull redis/redis-stack-server:$REDIS_STACK_VERSION
+
+    docker run --name redis --restart always -v $REDIS_CONF_FILE:/redis-stack.conf --port ${REDIS_PORT}:6379 -d redis/redis-stack-server:$REDIS_STACK_VERSION
+}
+
+user_setup
+docker_setup
+redis_setup
```
Comment (andre-senna) [inline install-redis.sh:132]: I didn't see where Redis port is being set open in ufw. Are we doing it somewhere else?
Anyway, this port should not be open for every IP, but only for the OpenFaaS server.
This setup is for standalone Redis only, right? If not I see some other missing steps here.

File: install-openfaas.sh [hunk 4]
```diff
@@ -13,33 +18,44 @@ cd faasd
...
+
+function firewall_setup() {
+    if ! command -v ufw &>/dev/null; then
+        apt-get update
+        apt-get install ufw
+    fi
+
+    ufw -y enable
+    ufw allow 8080/tcp
+    ufw allow ssh
```
Comment (andre-senna) [inline install-openfaas.sh:55]: SSH must be enabled in all servers. I'm not sure how you are enabling SSH so perhaps this command is redundant.

**Review by andre-senna — CHANGES_REQUESTED** [review]

File: install-redis.sh (hunk 3 above)
Comment (levisingularity) [reply install-redis.sh:164]: Yes, this is a standalone setup. I was considering initiating a new task to provision a Redis and MongoDB cluster using Terraform.

**Review by andre-senna — APPROVED** [review]

**Pull request merged**